crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -76,17 +78,60 @@ impl DynamicScene {
         scene
     }
 
+    /// Create a new dynamic scene from a given [`World`] at a given [`Entity`].
+    ///
+    /// All descendants entities from the given one will be extracted.
+    pub fn from_world_at_root(
+        world: &World,
+        root: Entity,
+        type_registry: &AppTypeRegistry,
+    ) -> Self {
+        let mut scene = DynamicScene::default();
+        let type_registry = type_registry.read();
+
+        // Extract all entities that are a descendant of root
+        let mut entities_of_interest = list_all_descendants(world, root);
+        entities_of_interest.push(root);
+
+        for entity in entities_of_interest {
+            scene.entities.push(DynamicEntity {
+                entity: entity.id(),
+                components: Vec::new(),
+            });
+
+            for component_id in world.entity(entity).archetype().components() {
+                let reflect_component = world
+                    .components()
+                    .get_info(component_id)
+                    .and_then(|info| type_registry.get(info.type_id().unwrap()))
+                    .and_then(|registration| registration.data::<ReflectComponent>());
+                if let Some(reflect_component) = reflect_component {
+                    if let Some(component) = reflect_component.reflect(world, entity) {
+                        scene
+                            .entities
+                            .last_mut()
+                            .unwrap()
+                            .components
+                            .push(component.clone_value());
+                    }
+                }
+            }
+        }
+
+        scene
+    }
+
     /// Write the dynamic entities and their corresponding components to the given world.
     ///
     /// This method will return a [`SceneSpawnError`] if a type either is not registered
     /// or doesn't reflect the [`Component`](bevy_ecs::component::Component) trait.
-    pub fn write_to_world(
+    pub fn write_to_world_with(
```

Comment:
I think this should be a separate method so users that want to extract everything can don't need to add the extra `world.resource::<AppTypeRegistry>()` call.